### PR TITLE
Reactify and clean up controllers

### PR DESCRIPTION
**Pull request by philippotto**
This PR makes the root controller a react component and cleans up some bits and pieces around our controller achitecture. I'm not really sure how we want to proceed and thought it would be good to get this first pass reviewed.

### Steps to test:
- usual general testing

### Issues:
- contributes to #1858 

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/controller/annotations/skeletontracing_controller.js [hunk 1]
```diff
@@ -1,80 +0,0 @@
...
-
-
-  setRadius(delta: number): void {
-    getActiveNode(Store.getState().tracing)
-      .map(activeNode =>
-        Store.dispatch(setActiveNodeRadiusAction(activeNode.radius * Math.pow(1.05, delta))));
-  }
-
-
-  toggleSkeletonVisibility = (): void => {
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/controller/annotations/skeletontracing_controller.js]: As far as I can see, this code was never used, which is why I didn't bother to reincorporate this somewhere. We might want to keep this in mind.

Comment (daniel-wer) [inline app/assets/javascripts/oxalis/controller/annotations/skeletontracing_controller.js]: Same for setParticleSize right? Strange how this ended up there, remains from the earlier days :P
I know that showFirstVisToggle didn't work before your PR, but we should either fix this functionality again or remove it completely (together with the `firstVisToggle` user setting).

File: app/assets/javascripts/oxalis/api/api_latest.js [hunk 2]
```diff
@@ -229,6 +230,32 @@ class TracingApi {
...
+  //  SKELETONTRACING API
+
+  /**
+   * Increases the node radius of the active node by multiplying it with 1.05^delta.
+   *
+   * @example
+   * api.tracing.setRadius(1)
+   */
+  setRadius(delta: number): void {
+    getActiveNode(Store.getState().tracing)
```
Comment (normanrz) [inline app/assets/javascripts/oxalis/api/api_latest.js]: As I already mentioned in our meeting on Weds, I'd like the API to be more generic, i.e. allow to specify specific nodes wherever possible and default to activeNode. The same for edges, trees etc.

Something like:
```
setNodeRadiusByDelta(delta: number, nodeId?: number, treeId?: number): void {
  getNodeAndTree(Store.getState().tracing, nodeId, treeId).
    map(([tree, node]) => Store.dispatch(setNodeRadiusAction(node.id, tree.id, node.radius * Math.pow(1.05, delta)))));
```

File: app/assets/javascripts/oxalis/controller/annotations/skeletontracing_controller.js (hunk 1 above)
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/controller/annotations/skeletontracing_controller.js]: This got lost when doing the initial Reduxify PR. There is a follow up issues to restore this "firstVisible" warning.  Perhaps, we can quickly add this here and resolve this once and for all. https://github.com/scalableminds/webknossos/issues/1762

File: app/assets/javascripts/oxalis/view/tracing_layout_view.js [hunk 3]
```diff
@@ -72,42 +67,51 @@ class TracingLayoutView extends React.PureComponent {
...
-                    visible={this.state.isUserScriptsModalOpen}
-                    onClose={this.closeUserScriptsModal}
-                  />
-                  <TracingView />
-                </div>
-                <div style={{ flex: "1", display: "inline-flex" }}>
-                  <RightMenuView />
+          <div>
+            <OxalisController
+              initialTracingType={this.props.initialTracingType}
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/view/tracing_layout_view.js]: Nice.

File: app/assets/javascripts/oxalis/api/api_latest.js [hunk 4]
```diff
@@ -238,27 +238,33 @@ class TracingApi {
...
-   * api.tracing.setRadius(1)
+   * api.tracing.setNodeRadius(1)
    */
-  setRadius(delta: number): void {
-    getActiveNode(Store.getState().tracing)
-      .map(activeNode =>
-        Store.dispatch(setActiveNodeRadiusAction(activeNode.radius * Math.pow(1.05, delta))));
+  setNodeRadius(delta: number, nodeId?: number, treeId?: number): void {
+    getNodeAndTree(Store.getState().tracing, nodeId, treeId)
+      .map(([, node]) =>
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/api/api_latest.js]: no treeId?

**Pull request merged**